src/sumbuddy/__main__.py

```diff
@@ -9,24 +9,32 @@
 import sys
 import os
 
-def get_checksums(input_directory, output_filepath=None, ignore_file=None, include_hidden=False, algorithm='md5', length=None):
+def get_checksums(input_path, output_filepath=None, ignore_file=None, include_hidden=False, algorithm='md5', length=None):
     """
     Generate a CSV file with the filepath, filename, and checksum of all files in the input directory according to patterns to ignore. Checksum column is labeled by the selected algorithm (e.g., 'md5' or 'sha256').
     
     Parameters:
     ------------
-    input_directory - String. Directory to traverse for files.
+    input_path - String. File or directory to traverse for files.
     output_filepath - String [optional]. Filepath for the output CSV file. Defaults to None, i.e. output will be to stdout.
     ignore_file - String [optional]. Filepath for the ignore patterns file.
     include_hidden - Boolean [optional]. Whether to include hidden files. Default is False.
     algorithm - String. Algorithm to use for checksums. Default: 'md5', see options with 'hashlib.algorithms_available'.
     length - Integer [conditionally optional]. Length of the digest for SHAKE (required) and BLAKE (optional) algorithms in bytes.
     """
     mapper = Mapper()
-    try:
-        file_paths = mapper.gather_file_paths(input_directory, ignore_file=ignore_file, include_hidden=include_hidden)
-    except (EmptyInputDirectoryError, NoFilesAfterFilteringError) as e:
-        sys.exit(str(e))
+
+    if os.path.isfile(input_path):
+        file_paths = [input_path]
+        if ignore_file:
+            print("Warning: --ignore-file (-i) flag is ignored when input is a single file.")
+        if include_hidden:
+            print("Warning: --include-hidden (-H) flag is ignored when input is a single file.")
+    else:
+        try:
+            file_paths = mapper.gather_file_paths(input_path, ignore_file=ignore_file, include_hidden=include_hidden)
+        except (EmptyInputDirectoryError, NoFilesAfterFilteringError) as e:
+            sys.exit(str(e))
```

Comment:
The addition of this `isfile` condition block is all that's needed for core functionality of this PR.